### PR TITLE
refactor: replace breadcrumbs with new top navbar

### PR DESCRIPTION
**Pull request by ogzhanolguncu**
## What does this PR do?

Fixes # (issue)

*If there is not an issue for this, please create one first. This is used to tracking purposes and also helps use understand why this PR exists*

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Chore (refactoring code, technical debt, workflow improvements)
- [x] Enhancement (small improvements)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How should this be tested?

- Check any page that had breadcrumbs

### Required

- [ ] Filled out the "How to test" section in this PR
- [ ] Read [Contributing Guide](./CONTRIBUTING.md)
- [ ] Self-reviewed my own code
- [ ] Commented on my code in hard-to-understand areas
- [ ] Ran `pnpm build`
- [ ] Ran `pnpm fmt`
- [ ] Checked for warnings, there are none
- [ ] Removed all `console.logs`
- [ ] Merged the latest changes from main onto my branch with `git pull origin main`
- [ ] My changes don't cause any responsiveness issues

### Appreciated

- [ ] If a UI change was made: Added a screen recording or screenshots to this PR
- [ ] Updated the Unkey Docs if changes were necessary

## Summary by CodeRabbit

- **New Features**
	- Introduced a navigation constant for APIs and rate limits, enhancing user navigation.
	- Added a structured layout with a `Navbar` and `PageContent` for various settings and audit pages.
	- Integrated breadcrumbs for improved navigation across multiple components.

- **Bug Fixes**
	- Resolved issues with loading states and component rendering for various dashboard pages.

- **Refactor**
	- Streamlined component structures, reducing complexity and improving readability across multiple files.
	- Updated various components to use a consistent layout structure, enhancing user experience.

- **Chores**
	- Removed unused components and imports to clean up the codebase.

### CODE REVIEW

File: apps/dashboard/app/(app)/apis/[apiId]/keys/[keyAuthId]/new/page.tsx [hunk 1]
```diff
@@ -24,11 +29,45 @@ export default async function CreateKeypage(props: {
...
+          </Navbar.Breadcrumbs.Link>
+          <Navbar.Breadcrumbs.Link
+            active
+            href={`/apis/${props.params.apiId}/keys/${keyAuth.id}/new`}
+          >
+            Create new key
+          </Navbar.Breadcrumbs.Link>
+        </Navbar.Breadcrumbs>
+        <Navbar.Actions>
+          <div className="flex items-center gap-2">
```
Comment (chronark) [inline apps/dashboard/app/(app)/apis/[apiId]/keys/[keyAuthId]/new/page.tsx:50]: isn't the `Actions` supposed to take care of this styling?

File: apps/dashboard/app/(app)/ratelimits/[namespaceId]/logs/page.tsx [hunk 2]
```diff
@@ -67,25 +74,49 @@ export default async function AuditPage(props: Props) {
...
-        <Filters />
+      <Navbar>
+        <Navbar.Breadcrumbs icon={<Gauge />}>
+          <Navbar.Breadcrumbs.Link href="/ratelimits">Ratelimits</Navbar.Breadcrumbs.Link>
+          <Navbar.Breadcrumbs.Link href={`/ratelimits/${props.params.namespaceId}`} isIdentifier>
+            {namespace.name}
+          </Navbar.Breadcrumbs.Link>
+          <Navbar.Breadcrumbs.Link href="Logs" active>
+            Logs{" "}
+          </Navbar.Breadcrumbs.Link>
```
[the remark this answers is not in the document]
Comment (ogzhanolguncu) [reply apps/dashboard/app/(app)/ratelimits/[namespaceId]/logs/page.tsx:85]: Good catch.

File: apps/dashboard/app/(app)/apis/[apiId]/keys/[keyAuthId]/page.tsx [hunk 3]
```diff
@@ -28,8 +36,39 @@ export default async function APIKeysPage(props: {
...
+          <Navbar.Breadcrumbs.Link href="/apis">APIs</Navbar.Breadcrumbs.Link>
+          <Navbar.Breadcrumbs.Link href={`/apis/${props.params.apiId}`} isIdentifier>
+            {keyAuth.api.name}
+          </Navbar.Breadcrumbs.Link>
+          <Navbar.Breadcrumbs.Link active href={`/apis/${props.params.apiId}/keys/${keyAuth.id}`}>
+            Keys
+          </Navbar.Breadcrumbs.Link>
+        </Navbar.Breadcrumbs>
+        <Navbar.Actions>
+          <div className="flex items-center gap-2">
```
Comment (chronark) [inline apps/dashboard/app/(app)/apis/[apiId]/keys/[keyAuthId]/page.tsx:51]: same as above

File: apps/dashboard/app/(app)/apis/[apiId]/layout.tsx
Comment (chronark) [inline apps/dashboard/app/(app)/apis/[apiId]/layout.tsx:1]: I think we can just remove this file now
it doesn't do anything

File: apps/dashboard/app/(app)/apis/[apiId]/page.tsx [hunk 4]
```diff
@@ -129,138 +141,169 @@ export default async function ApiPage(props: {
...
+    <div>
+      <Navbar>
+        <Navbar.Breadcrumbs icon={<Nodes />}>
+          <Navbar.Breadcrumbs.Link href="/apis">APIs</Navbar.Breadcrumbs.Link>
+          <Navbar.Breadcrumbs.Link href={`/apis/${props.params.apiId}`} active isIdentifier>
+            {api.name}
+          </Navbar.Breadcrumbs.Link>
+        </Navbar.Breadcrumbs>
+        <Navbar.Actions>
+          <div className="flex items-center gap-2">
```
Comment (chronark) [inline apps/dashboard/app/(app)/apis/[apiId]/page.tsx:153]: move styling into actions

File: apps/dashboard/app/(app)/apis/[apiId]/settings/page.tsx [hunk 5]
```diff
@@ -49,27 +56,58 @@ export default async function SettingsPage(props: Props) {
...
+          <Navbar.Breadcrumbs.Link href="/apis">APIs</Navbar.Breadcrumbs.Link>
+          <Navbar.Breadcrumbs.Link href={`/apis/${props.params.apiId}`} isIdentifier>
+            {api.name}
+          </Navbar.Breadcrumbs.Link>
+          <Navbar.Breadcrumbs.Link active href={`/apis/${props.params.apiId}/settings`}>
+            Settings
+          </Navbar.Breadcrumbs.Link>
+        </Navbar.Breadcrumbs>
+        <Navbar.Actions>
+          <div className="flex items-center gap-2">
```
Comment (chronark) [inline apps/dashboard/app/(app)/apis/[apiId]/settings/page.tsx:71]: move styling into actions

File: apps/dashboard/app/(app)/authorization/layout.tsx
Comment (chronark) [inline apps/dashboard/app/(app)/authorization/layout.tsx:1]: no longer needed

File: apps/dashboard/app/(app)/authorization/permissions/[permissionId]/page.tsx [hunk 6]
```diff
@@ -73,83 +69,96 @@ export default async function RolesPage(props: Props) {
...
-              <TooltipTrigger asChild>
-                <div className="flex items-center justify-between gap-2 w-full truncate">
-                  <span className="truncate">{permission.name}</span>
-                  <div>
-                    <CopyButton value={permission.name} />
+            {props.params.permissionId}
+          </Navbar.Breadcrumbs.Link>
+        </Navbar.Breadcrumbs>
+        <Navbar.Actions>
+          <div className="flex items-center gap-2">
```
Comment (chronark) [inline apps/dashboard/app/(app)/authorization/permissions/[permissionId]/page.tsx:91]: move styling into actions

File: apps/dashboard/app/(app)/authorization/permissions/page.tsx [hunk 7]
```diff
@@ -51,61 +56,89 @@ export default async function RolesPage() {
...
+        <Navbar.Breadcrumbs icon={<ShieldKey />}>
+          <Navbar.Breadcrumbs.Link href="/authorization/roles">
+            Authorization
+          </Navbar.Breadcrumbs.Link>
+          <Navbar.Breadcrumbs.Link href="/authorization/permissions" active>
+            Permissions
+          </Navbar.Breadcrumbs.Link>
+        </Navbar.Breadcrumbs>
+        <Navbar.Actions>
+          <div className="flex items-center gap-2">
```
Comment (chronark) [inline apps/dashboard/app/(app)/authorization/permissions/page.tsx:70]: move styling into actions

File: apps/dashboard/app/(app)/authorization/roles/[roleId]/page.tsx [hunk 8]
```diff
@@ -106,23 +109,47 @@ export default async function RolesPage(props: Props) {
...
+            href={`/authorization/permissions/${props.params.roleId}`}
+            isIdentifier
+            active
+            className="truncate w-[100px]"
+          >
+            {props.params.roleId}
+          </Navbar.Breadcrumbs.Link>
+        </Navbar.Breadcrumbs>
+        <Navbar.Actions>
           <div className="flex items-center gap-2">
```
Comment (chronark) [inline apps/dashboard/app/(app)/authorization/roles/[roleId]/page.tsx:129]: move styling into actions

File: apps/dashboard/app/(app)/authorization/roles/page.tsx [hunk 9]
```diff
@@ -51,70 +56,91 @@ export default async function RolesPage() {
...
+        <Navbar.Breadcrumbs icon={<ShieldKey />}>
+          <Navbar.Breadcrumbs.Link href="/authorization/roles">
+            Authorization
+          </Navbar.Breadcrumbs.Link>
+          <Navbar.Breadcrumbs.Link href="/authorization/roles" active>
+            Roles
+          </Navbar.Breadcrumbs.Link>
+        </Navbar.Breadcrumbs>
+        <Navbar.Actions>
+          <div className="flex items-center gap-2">
```
Comment (chronark) [inline apps/dashboard/app/(app)/authorization/roles/page.tsx:70]: move styling into actions

File: apps/dashboard/app/(app)/ratelimits/[namespaceId]/layout.tsx
Comment (chronark) [inline apps/dashboard/app/(app)/ratelimits/[namespaceId]/layout.tsx:1]: doesn't do anything

File: apps/dashboard/app/(app)/workspace-navigations.tsx [hunk 10]
```diff
@@ -121,7 +124,7 @@ export const createWorkspaceNavigation = (
       href: "/identities",
       label: "Identities",
       active: segments.at(0) === "identities",
-      hidden: !workspace.betaFeatures.identities,
```
Comment (chronark) [inline apps/dashboard/app/(app)/workspace-navigations.tsx:124]: why?
this is very much intended to be hidden

**Review by chronark — CHANGES_REQUESTED** [review]

File: apps/dashboard/app/(app)/workspace-navigations.tsx (hunk 10 above)
Comment (ogzhanolguncu) [reply apps/dashboard/app/(app)/workspace-navigations.tsx:124]: My bad. I had to enable it to make sure everything works.

**Comment by chronark** [comment]
CleanShot 2024-12-16 at 12 17 50@2x
https://dashboard-d0bwz9jze-unkey.vercel.app/apis/api_2w8vSkYnfLk4NgEBZqQX36t6Rt7i/keys/key_auth_2w8vRgTrd7AszbC6STEm8Sfj2FQm/key_4ZHZcXrpVBKxHX38RdAYMFH2tzs7

The breadcrumbs should be:
`APIs / scenario-test-delete-pls / ... / key_4ZHZ... `

we'll remove some breadcrumbs from the middle, rather than the end

**Comment by ogzhanolguncu** [comment]
> CleanShot 2024-12-16 at 12 17 50@2x https://dashboard-d0bwz9jze-unkey.vercel.app/apis/api_2w8vSkYnfLk4NgEBZqQX36t6Rt7i/keys/key_auth_2w8vRgTrd7AszbC6STEm8Sfj2FQm/key_4ZHZcXrpVBKxHX38RdAYMFH2tzs7
> 
> The breadcrumbs should be: `APIs / scenario-test-delete-pls / ... / key_4ZHZ... `
> 
> we'll remove some breadcrumbs from the middle, rather than the end
> 
> @ogzhanolguncu
> 
> same here: https://dashboard-d0bwz9jze-unkey.vercel.app/apis/api_2w8vSkYnfLk4NgEBZqQX36t6Rt7i/keys/key_auth_2w8vRgTrd7AszbC6STEm8Sfj2FQm/key_4ZHZcXrpVBKxHX38RdAYMFH2tzs7/settings
> 
> keep the beginning and end

I thought about that then decided not to do it because there is plenty of space on the screen. I think we can keep it as is, wdyt?

**Comment by chronark** [comment]
> > CleanShot 2024-12-16 at 12 17 50@2x https://dashboard-d0bwz9jze-unkey.vercel.app/apis/api_2w8vSkYnfLk4NgEBZqQX36t6Rt7i/keys/key_auth_2w8vRgTrd7AszbC6STEm8Sfj2FQm/key_4ZHZcXrpVBKxHX38RdAYMFH2tzs7
> > The breadcrumbs should be: `APIs / scenario-test-delete-pls / ... / key_4ZHZ... `
> > we'll remove some breadcrumbs from the middle, rather than the end
> > @ogzhanolguncu
> > same here: https://dashboard-d0bwz9jze-unkey.vercel.app/apis/api_2w8vSkYnfLk4NgEBZqQX36t6Rt7i/keys/key_auth_2w8vRgTrd7AszbC6STEm8Sfj2FQm/key_4ZHZcXrpVBKxHX38RdAYMFH2tzs7/settings
> > keep the beginning and end
> 
> I thought about that then decided not to do it because there is plenty of space on the screen. I think we can keep it as is, wdyt?

it's not about space on the screen, it's about showing the most important parts

**Review by chronark — APPROVED** [review]

**Pull request merged**